### PR TITLE
(fix) - replaceNode should unmount when diffing twice with replaceNode

### PR DESCRIPTION
**Pull request by JoviDeCroock**
Fixes: https://github.com/preactjs/preact/issues/1722

Adds 7Bytes

### CODE REVIEW

File: src/render.js [hunk 1]
```diff
@@ -13,13 +13,16 @@ import options from './options';
  */
 export function render(vnode, parentDom, replaceNode) {
 	if (options._root) options._root(vnode, parentDom);
-	let oldVNode = parentDom._children;
+	let oldVNode = replaceNode ?
+		replaceNode._children || parentDom._children :
```
Comment (JoviDeCroock) [inline src/render.js]: This `||` is needed for the case when we first did `render(x, y, undefined)` and transition to a render with replaceNode. The second is needed for when we use replaceNode twice in a row.

**Comment by JoviDeCroock** [comment]
@cristianbote you are right, it seems to be after the most recent master merge, probably something that can be gzipped more efficiently since it occurs more.

File: src/render.js (hunk 1 above)
Comment (developit) [reply src/render.js]: What about this?

```suggestion
replaceNode && replaceNode._children || parentDom._children
```

File: src/render.js [hunk 2]
```diff
@@ -13,13 +13,16 @@ import options from './options';
...
+		parentDom._children;
+
 	vnode = createElement(Fragment, null, [vnode]);
 
 	let mounts = [];
 	diff(
 		parentDom,
-		replaceNode ? vnode : (parentDom._children = vnode),
+		replaceNode ? (replaceNode._children = vnode) : (parentDom._children = vnode),
```
Comment (developit) [inline src/render.js]: Possibly smaller:

```suggestion
		(replaceNode || parentDom)._children = vnode,
```

**Pull request merged**